apps/api-runtime/src/main/java/com/akto/runtime/APICatalogSync.java

```diff
@@ -1079,7 +1063,18 @@ public DbUpdateReturn getDBUpdatesForParams(APICatalog currentDelta, APICatalog
             }
 
 
-            if (!redactSampleData && deltaInfo.getExamples() != null && !deltaInfo.getExamples().isEmpty()) {
+            if (!(redactSampleData || collectionLevelRedact) && deltaInfo.getExamples() != null && !deltaInfo.getExamples().isEmpty()) {
+                Set<Object> updatedSampleData = new HashSet<>();
+                for (Object example : deltaInfo.getExamples()) {
+                    try{
+                        String exampleStr = (String) example;
+                        String s = RedactSampleData.redactMarkedDataTypes(exampleStr);
```

Comment:
This should be called at exactly one place in the entire code. Why we need at 2 places?

Reply:
Here we are doing it for sensitive sample data and in getDBUpdatesForSampleData method its for sample data